x-pack/plugins/security_solution/server/deprecation_privileges/index.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+
+import { DeprecationsDetails, DeprecationsServiceSetup } from '../../../../../src/core/server';
+import type { PrivilegeDeprecationsServices } from '../../../security/common/model';
+import { CASES_FEATURE_ID, SERVER_APP_ID } from '../../common/constants';
+
+interface Deps {
+  deprecationsService: DeprecationsServiceSetup;
+  getKibanaRolesByFeatureId?: PrivilegeDeprecationsServices['getKibanaRolesByFeatureId'];
+}
+
+export const updateSecuritySolutionPrivileges = (
+  siemPrivileges: string[]
+): Partial<Record<typeof SERVER_APP_ID | typeof CASES_FEATURE_ID, string[]>> => {
+  const siemPrivs = new Set<string>();
+  const casesPrivs = new Set<string>();
+
+  for (const priv of siemPrivileges) {
```

Comment:
Thank you @jonathan-buttner, your idea to use a switch and a set make it more readable and easier to understand. Thanks for the suggestion.